magma/when.py

```diff
@@ -526,6 +546,108 @@ def find_inferred_latches(block: _BlockBase) -> Set:
     return latches
 
 
+# Use a global counter so assertion names are unique across the design
+# (not just per module)
+_WHEN_ASSERT_COUNTER = itertools.count()
```

Comment:
we should just use `assert` or `when_assert` rather than mix the two. locally to this file can just be `assert`